src/serializer/ResidualHeapSerializer.js

```diff
@@ -557,15 +557,17 @@ export class ResidualHeapSerializer {
 
   // Determine whether initialization code for a value should go into the main body, or a more specific initialization body.
   _getTarget(
-    val: Value,
-    scopes: Set<Scope>
+    val: Value
   ): {
     body: SerializedBody,
     usedOnlyByResidualFunctions?: true,
     usedOnlyByAdditionalFunctions?: boolean,
     commonAncestor?: Scope,
     description?: string,
   } {
+    let scopes = this.residualValues.get(val);
+    invariant(scopes !== undefined);
```

Comment:
It will help to reader if you explain in a code comment why you believe the invariant to be true.